src/rnp/fficli.h

```diff
@@ -72,8 +81,42 @@ void cli_rnp_print_key_info(
   FILE *fp, rnp_ffi_t ffi, rnp_key_handle_t key, bool psecret, bool psigs);
 bool cli_rnp_set_generate_params(rnp_cfg_t *cfg);
 bool cli_rnp_generate_key(rnp_cfg_t *cfg, cli_rnp_t *rnp, const char *username);
-list cli_rnp_get_keylist(cli_rnp_t *rnp, const char *filter, bool secret, bool subkeys);
-void cli_rnp_keylist_destroy(list *keys);
+/**
+ * @brief Find key(s) matching set of flags and search string.
+ *
+ * @param rnp initialized cli_rnp_t object.
+ * @param keys search results will be added here, leaving already existing items.
+ * @param str search string: may be part of the userid, keyid, fingerprint or grip.
+ * @param flags combination of the following flags:
+ *              CLI_SEARCH_SECRET : require key to be secret,
+ *              CLI_SEARCH_SUBKEYS : include subkeys to the results (see
+ *                CLI_SEARCH_SUBKEYS_AFTER description).
+ *              CLI_SEARCH_FIRST_ONLY : include only first key found
+ *              CLI_SEARCH_SUBKEYS_AFTER : for each primary key add it's subkeys after the main
```

Comment:
@rrrooommmaaa Ok, added phrase about the orphaned subkeys. This code is mostly internal, and could be subject to changes, so it is not worth to heavily document it.
And the main idea of this PR is to move whole key search functionality to the single function so it would be reused in functionality, added later.

Reply:
Since this is an internal function, is documented, and does appear to accomplish the task in the title of the PR, I'm OK with this as-is. `key_matches_flags` is an internal helper, we can always add a comment to explain any potentially non-obvious behavior.